File: src/decoders/new-discussion-recipient-subject.ts
import type { NewDiscussionRecipientSubSubject, NewDiscussionRecipientSubject } from "~/models";

export const decodeNewDiscussionRecipientSubject = (subject: any, sub: Array<NewDiscussionRecipientSubSubject>): NewDiscussionRecipientSubject => {
  return {
    id: subject.N,
    name: subject.L,
    sub: sub
  };
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/decoders/new-discussion-recipient-subject.kt
package ink.literate.pawnote.decoders

import ink.literate.pawnote.models.NewDiscussionRecipientSubSubject
import ink.literate.pawnote.models.NewDiscussionRecipientSubject
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonPrimitive

fun decodeNewDiscussionRecipientSubject (subject: JsonObject, sub: List<NewDiscussionRecipientSubSubject>) = NewDiscussionRecipientSubject(
    id = subject["N"]!!.jsonPrimitive.content,
    name = subject["L"]!!.jsonPrimitive.content,
    sub = sub
)